README.md

```diff
@@ -0,0 +1,65 @@
+# SaplingMcp
+
+A Model Context Protocol (MCP) server for Sapling SCM integration.
+
+## Token-Efficient Parsing Format
+
+This project implements a token-efficient line-based parsing format for the MCP server that handles stack information and PR comments for Sapling SCM. The format is designed to be compact and efficient when interacting with LLMs.
+
+### Stack Information Format
+
+Stack information is formatted using a pipe-delimited structure:
+
+```
+sha:<commit-sha>|title:<commit-title>|pr:<owner/repo#number or none>
+```
+
+Example:
+```
+sha:abc123def456|title:Fix widget rendering bug|pr:your-org/your-repo#123
+```
+
+### PR Comments Format
+
+PR comments are formatted with a similar pipe-delimited structure:
+
+```
+pr:<owner/repo#number>|author:<username>|date:<timestamp>|id:<comment-id>|body:<comment-text>
+```
+
+Example:
+```
+pr:your-org/your-repo#123|author:username|date:2025-04-20T14:30:00Z|id:comment123|body:This looks good.\nJust one suggestion: let's add more tests.
+```
+
+### Special Characters
+
+- Newlines in comments are escaped as `\\n` to maintain the line-based format
+- Each item is on a separate line, making it easy to process line by line
+
+### Usage
+
+The MCP server provides token-efficient versions of the following endpoints:
+
+- `GetCurrentStackTokenEfficient`: Gets all commits in the current stack in token-efficient format
+- `GetPublicCommitsTokenEfficient`: Gets all public commits in token-efficient format
+- `GetPullRequestCommentsTokenEfficient`: Gets comments for a specific pull request in token-efficient format
```

Comment:
In don't think these methods are defined any more. Can you check and update the read me as required?